File: apps/push-service/src/main/java/com/example/chat/push/domain/PushMessage.java
package com.example.chat.push.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * 푸시 메시지 엔티티
 */
@Entity
@Table(name = "push_messages")
@Getter
@NoArgsConstructor(access = lombok.AccessLevel.PROTECTED)
public class PushMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String targetUserId;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Setter
    private PushStatus status = PushStatus.PENDING;

    @Column(nullable = false)
    private String pushType;

    @Column(nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column
    @Setter
    private LocalDateTime processedAt;

    @Column
    @Setter
    private String errorMessage;

    @Version
    @Setter
    private Long version = 0L;

    private PushMessage(String targetUserId, String title, String content, String pushType) {
        this.targetUserId = targetUserId;
        this.title = title;
        this.content = content;
        this.pushType = pushType;
    }

    public static PushMessage of(String targetUserId, String title, String content, String pushType) {
        return new PushMessage(targetUserId, title, content, pushType);
    }

    public void markProcessing() {
        this.status = PushStatus.PROCESSING;
    }

    public void markCompleted() {
        this.status = PushStatus.COMPLETED;
        this.processedAt = LocalDateTime.now();
    }

    public void markFailed(String errorMessage) {
        this.status = PushStatus.FAILED;
        this.errorMessage = errorMessage;
    }
}

File: apps/push-service/src/main/kotlin/com/example/chat/push/domain/PushMessage.kt
package com.example.chat.push.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "push_messages")
class PushMessage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val targetUserId: String,

    @Column(nullable = false)
    val title: String,

    @Column(nullable = false, columnDefinition = "TEXT")
    val content: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: PushStatus = PushStatus.PENDING,

    @Column(nullable = false)
    val pushType: String, // e.g., TOAST, FCM, SLACK

    @Column(nullable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @Column
    var processedAt: LocalDateTime? = null,

    @Column
    var errorMessage: String? = null,

    @Version
    var version: Long = 0
)

enum class PushStatus {
    PENDING, PROCESSING, COMPLETED, FAILED
}
